ipakyulibank/actions
# Path filter & Notify actions

## Description
Filtering paths when a release occurs

## What is done
#### **1. Path-filter action principle of operation**
  - Check eventType
    - If "release"
      - Checks the event type
      - Checks if the action has been issued
      - Checks if the action is already "prerelease"
      - Checks if filter_string is specified in ENV
      - Find out latest releases with github api
      - Pull repo with tags
      - Compares required tags with "git diff --names-only"
    - If "pull_request" or "push"
      - Checks if filter_string is specified in ENV
      - Make sure the base and head properties are set for the payload
      - The name of the diff files is loaded with github rest api comparing base and head refs
  - If there is at least one file in the list of files that satisfies the condition in the filter, it will return result = '1' to the variable, otherwise '0'
#### **2. Notify action principle of operation**
  - Checks the length of the BODY parameter
    - If the length of the BODY parameter exceeds 3990
      - will return an exception
    - sends a notification
  - The result is returned as '1' or '0' when the notification is sent
